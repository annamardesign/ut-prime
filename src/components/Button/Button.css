.btn {
    text-transform: uppercase;
    font-family: OpenSans;
    font-size: 14px;
    color: #FFFFFF;
    border-radius: 2px;
    border: 0px;
    font-weight: 600;
}

.btn:focus {
    outline: #014c8c;
    outline: 1px dotted;
    outline: 5px auto -webkit-focus-ring-color;
}

.btnNormal {
    width: 146px;
    height: 46px;
}

.btnSmall {
    font-size: 11px;
    line-height: 11px;
    padding: 8px 10px;
}

.btnFullWidth {
    width: 100%;
}

.btn:disabled {
    background-color: #F2F2F2;
    cursor: pointer;
}

.btn:disabled:hover {
    background-color: #F2F2F2;
    cursor: pointer;
}

.btnGreen {
    background-color: #7EC241;
}

.btnGreen:hover {
    background-color: #72A73F;
}

.btnPink {
    background-color: #0074ba;
}

.btnPink:hover {
    background-color: #0067a6;
}

.btnGrey {
    background-color: #D2D5D6;
}

.btnGrey:hover {
    background-color: #777E7D;
}

.btnGreyDark {
    background-color: #4D525A;
}

.btnGreyDark:hover {
    background-color: #384044;
}

.btnClose {
    width: 15px;
    height: 15px;
    color: transparent;
    overflow: hidden;
    padding: 0px;
}

.btnClose:hover {
    width: 15px;
    height: 15px;
    color: transparent;
    overflow: hidden;
    padding: 0px;
}

.btnClosePage {
    width: 40px;
    height: 31px;
    color: transparent;
    overflow: hidden;
}

.btnClosePage:hover {
    font-weight: normal;
}

.btnValidateFile {
    padding-left: 30px;
    padding-right: 10px;
    background-color: #FFFFFF;
    border: 1px solid #7EC241;
    color: #7EC241;
}

.btnValidateFile:hover {
    background-color: #72A73F;
    color: #FFFFFF;
}

.btnRejectFile {
    padding-left: 30px;
    padding-right: 10px;
    background-color: #FFFFFF;
    border: 1px solid #EA4949;
    color: #EA4949;
}

.btnRejectFile:hover {
    background-color: #EA4949;
    color: #FFFFFF;
}

.btnCustom {
    background: none;
    border: none;
}
