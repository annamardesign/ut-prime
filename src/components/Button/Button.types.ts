import {WithStyles, withStyles, createStyles} from '@material-ui/core/styles';
import React from 'react';

export interface Props extends React.HTMLAttributes<HTMLButtonElement> {
    className?: string;
    button?: string;
    sizeType?: string;
    fullWidth?: boolean;
}

const styles = createStyles({
    btn: {
        textTransform: 'uppercase',
        fontSize: 14,
        color: '#FFFFFF',
        borderRadius: 2,
        border: 0,
        fontWeight: 600,
        '&:focus': {
            outline: '5px auto -webkit-focus-ring-color',
            fallbacks: {
                outline: '1px dotted #014c8c'
            }
        },
        '&:disabled': {
            backgroundColor: '#F2F2F2',
            cursor: 'pointer'
        },
        '&:disabled:hover': {
            backgroundColor: '#F2F2F2',
            cursor: 'pointer'
        }
    },
    btnNormal: {
        width: 146,
        height: 46
    },
    btnSmall: {
        fontSize: 11,
        lineHeight: '11px',
        padding: '8px 10px'
    },
    btnFullWidth: {
        width: '100%'
    },
    btnGreen: {
        backgroundColor: '#7EC241',
        '&:hover': {
            backgroundColor: '#72A73F'
        }
    },
    btnPink: {
        backgroundColor: '#0074ba',
        '&:hover': {
            backgroundColor: '#0067a6'
        }
    },
    btnGrey: {
        backgroundColor: '#D2D5D6',
        '&:hover': {
            backgroundColor: '#777E7D'
        }
    },
    btnGreyDark: {
        backgroundColor: '#4D525A',
        '&:hover': {
            backgroundColor: '#384044'
        }
    },
    btnClose: {
        width: 15,
        height: 15,
        color: 'transparent',
        overflow: 'hidden',
        padding: 0,
        '&:hover': {
            width: 15,
            height: 15,
            color: 'transparent',
            overflow: 'hidden',
            padding: 0
        }
    },
    btnClosePage: {
        width: 40,
        height: 31,
        color: 'transparent',
        overflow: 'hidden',
        '&:hover': {
            fontWeight: 'normal'
        }
    },
    btnValidateFile: {
        paddingLeft: 30,
        paddingRight: 10,
        backgroundColor: '#FFFFFF',
        border: '1px solid #7EC241',
        color: '#7EC241',
        '&:hover': {
            backgroundColor: '#72A73F',
            color: '#FFFFFF'
        }
    },
    btnRejectFile: {
        paddingLeft: 30,
        paddingRight: 10,
        backgroundColor: '#FFFFFF',
        border: '1px solid #EA4949',
        color: '#EA4949',
        '&:hover': {
            backgroundColor: '#EA4949',
            color: '#FFFFFF'
        }
    },
    btnCustom: {
        background: 'none',
        border: 'none'
    }
});

export const Styled = withStyles(styles);
export type StyledType = React.FC<Props & WithStyles<typeof styles>>
