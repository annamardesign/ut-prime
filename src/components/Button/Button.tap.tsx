import React from 'react';
import { render } from '../test';
import { fireEvent } from '@testing-library/react';

import Button from './index';
import tap from 'tap';

tap.test('<Button />', test => {
    test(assert => {
        const { getByTestId } = render(<Button />);
        assert.matchSnapshot(getByTestId('Button'), 'Button');
    }, 'renders component without break');

    test(assert => {
        let clicks = 0;
        const onClick = () => clicks++;
        const { getByTestId } = render(<Button onClick={onClick} />);
        fireEvent.click(getByTestId('Button'));
        assert(clicks, 1, 'Click was called');
    }, 'triggers sent onClick function');
});
