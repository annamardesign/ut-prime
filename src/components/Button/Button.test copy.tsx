import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';
import tap from 'tap';

tap('<Button />', test => {
    test('renders component without break', assert => {
        const { getByTestId } = render(<Button data-testid="Button" />);
        assert.matchSnapshot(getByTestId('Button'), 'Button');
    });

    test('triggers sent onClick function', assert => {
        let clicks = 0;
        const onClick = () => clicks++;
        const { getByTestId } = render(<button data-testid="Button" onClick={onClick} />);
        fireEvent.click(getByTestId('Button'));
        assert(clicks, 1, 'Click was called');
    });
});
