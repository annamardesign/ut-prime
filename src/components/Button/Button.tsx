import React from 'react';
import classnames from 'classnames';

import { IButtonProps } from './Button.types';
import style from './Button.css';

const Button: React.FC<IButtonProps> = ({ className, children, button, sizeType, fullWidth = false, ...props }) => {
    function getStyle(name) {
        // todo theming
        // return (this.context.implementationStyle && this.context.implementationStyle[name]) || style[name];
        return style[name];
    }

    let sizeTypeClass;

    switch (sizeType) {
        case 'small':
            sizeTypeClass = getStyle('btnSmall');
            break;
        default:
            sizeTypeClass = getStyle('btnNormal');
    }
    const fullWidthClass = fullWidth ? getStyle('btnFullWidth') : null;
    let typeClass;
    switch (button) {
        case 'add':
        case 'import':
        case 'save':
        case 'send':
        case 'validate':
            typeClass = getStyle('btnGreen');
            break;
        case 'back':
        case 'cancel':
        case 'reset':
        case 'retake':
            typeClass = getStyle('btnGrey');
            break;
        case 'connection':
        case 'next':
            typeClass = getStyle('btnPink');
            break;
        case 'end':
            typeClass = getStyle('btnGreyDark');
            break;
        case 'validate-file':
            typeClass = getStyle('btnValidateFile');
            break;
        case 'reject-file':
            typeClass = getStyle('btnRejectFile');
            break;
        case 'close':
            typeClass = getStyle('btnClose');
            break;
        case 'close-page':
            typeClass = getStyle('btnClosePage');
            break;
        case 'custom':
            typeClass = getStyle('btnCustom');
            break;
        default:
            typeClass = getStyle('btnGreen');
    }
    let classes = classnames(getStyle('btn'), sizeTypeClass, typeClass, fullWidthClass);
    if (button === 'custom') {
        classes = typeClass;
    }
    return (
        <button type='button' className={classes} tabIndex={0} {...props}>{children}</button>
    );
};

export default Button;
