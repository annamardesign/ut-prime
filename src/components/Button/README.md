# Button

## How to use

```jsx
import Button from 'ut-front-devextreme/core/Button';

<Button onClick={handlerOnClick}>Hello Button</Button>;
```

## Props

- **className** - class applied to root `Button` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
