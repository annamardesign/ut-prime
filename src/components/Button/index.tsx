import React from 'react';
import clsx from 'clsx';

import { useStyles, ComponentProps } from './Button.types';

const Button: ComponentProps = ({ children, button, sizeType, fullWidth = false, ...props }) => {
    let classes = useStyles();
    let sizeTypeClass;

    switch (sizeType) {
        case 'small':
            sizeTypeClass = classes.btnSmall;
            break;
        default:
            sizeTypeClass = classes.btnNormal;
    }
    const fullWidthClass = fullWidth ? classes.btnFullWidth : null;
    let typeClass;
    switch (button) {
        case 'add':
        case 'import':
        case 'save':
        case 'send':
        case 'validate':
            typeClass = classes.btnGreen;
            break;
        case 'back':
        case 'cancel':
        case 'reset':
        case 'retake':
            typeClass = classes.btnGrey;
            break;
        case 'connection':
        case 'next':
            typeClass = classes.btnPink;
            break;
        case 'end':
            typeClass = classes.btnGreyDark;
            break;
        case 'validate-file':
            typeClass = classes.btnValidateFile;
            break;
        case 'reject-file':
            typeClass = classes.btnRejectFile;
            break;
        case 'close':
            typeClass = classes.btnClose;
            break;
        case 'close-page':
            typeClass = classes.btnClosePage;
            break;
        case 'custom':
            typeClass = classes.btnCustom;
            break;
        default:
            typeClass = classes.btnGreen;
    }
    if (button === 'custom') {
        classes = typeClass;
    }
    return (
        <button type='button' className={clsx(classes.btn, sizeTypeClass, typeClass, fullWidthClass)} tabIndex={0} {...props}>{children}</button>
    );
};

export default Button;
