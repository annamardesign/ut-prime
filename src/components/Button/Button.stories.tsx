import React from 'react';
import { withReadme } from 'storybook-readme';
import { action } from '@storybook/addon-actions';

// @ts-ignore: md file and not a module
import README from './README.md';
import Button from './index';

export default {
    title: 'Button',
    component: Button,
    decorators: [withReadme(README)]
};

export const Basic: React.FC<{}> = () => <Button onClick={action('clicked')}>Hello Button</Button>;
