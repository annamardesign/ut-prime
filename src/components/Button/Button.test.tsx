import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Button from './index';

describe('<Button />', () => {
    it('render equals snapshot', () => {
        const { getByTestId } = render(<Button data-testid="Button" />);

        expect(getByTestId('Button')).toMatchSnapshot();
    });

    it('triggers sent onClick function', () => {
        const onClick = jest.fn();

        const { getByTestId } = render(<button data-testid="Button" onClick={onClick} />);

        fireEvent.click(getByTestId('Button'));

        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
