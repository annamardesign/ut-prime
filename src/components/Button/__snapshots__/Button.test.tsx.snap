// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render equals snapshot 1`] = `
<button
  class="Button-btn-1 Button-btnNormal-3 Button-btnGreen-8"
  data-testid="Button"
  tabindex="0"
  type="button"
/>
`;
